api/pkg/apis/projectcalico/v3/felixconfig.go

```diff
@@ -281,7 +281,11 @@ type FelixConfigurationSpec struct {
 
 	// This is the source address to use on programmed device routes. By default the source address is left blank,
 	// leaving the kernel to choose the source address used.
-	DeviceRouteSourceAddress string `json:"deviceRouteSourceAddress,omitempty"`
+	DeviceRouteV4SourceAddress string `json:"deviceRouteV4SourceAddress,omitempty"`
```

Comment:
Unfortunately, you can't just change the name of a field here.  It's part of our public API.  Think the best option is to leave the old name as-is and add the new one with an `IPv6` suffix.  Alternatively, you could adjust the syntax of the field to allow for `"<IPv4 address>[,<IPv6 address>]"`  Neither option is great!